### DISCUSSION TITLE
Issue with upgrade from 2.0.1 to 2.5.3

### DISCUSSION THREAD

**Discussion by cah-peter-g** [apache/airflow#32601] [Q&A]
Hello Community,

We are trying to upgrade Airflow from 2.0.1 to 2.5.3 but facing several issue with database, when we are tryin got import it from old db to new db we are facing below error as the DAG table don't have the compatibility issue, DAG warning has utf_8 charset abd ag table have nothing

_mysql.connection.query(self, query)
sqlalchemy.exc.OperationalError: (MySQLdb.OperationalError) (3780, "Referencing column 'dag_id' and referenced column 'dag_id' in foreign key constraint 'dcw_dag_id_fkey' are incompatible.")
[SQL:
CREATE TABLE dag_warning (
        dag_id VARCHAR(250) COLLATE utf8mb3_bin NOT NULL,
        warning_type VARCHAR(50) NOT NULL,
        message TEXT NOT NULL,
        timestamp TIMESTAMP(6) NOT NULL,
        CONSTRAINT dag_warning_pkey PRIMARY KEY (dag_id, warning_type),
        CONSTRAINT dcw_dag_id_fkey FOREIGN KEY(dag_id) REFERENCES dag (dag_id) ON DELETE CASCADE
)

Kindly suggest with your thoughts,

**Comment by potiuk** [comment]
I suggest to drop the old table. There was a problematic behaviour with MySQL encoding that caused all kinds of problems - especailly when migrating to older versions. MySql Changed utf8 meaning between different versions - specifically with encoding of the IDs of Airlfow. Also you have not specified which version of MySQL you have but, which also might contribute to the fact of having problems. Airlfow 2.5 will only work with 5.7 and 8 but if your DB migrated from 5.6 in the past, then you will have similar problems.

This isr really  MySQL problem and several people had it - since them we suggested many of them to migrate to Postgres - using MySQL and them changing defaults and behavior between versions was the root cause ot it and we could not really implement automation for all cases like that 

You have three choices:

1) drop your historical data (your dags will remain, you will just loose the history of executions. you wil  have to likely manually import/export configurations, variables, connections or the like

2) manually fix all the ids in your export - you can look at some of the issues people had in the past and likely you will find a manual process how some of them coped with it.
https://github.com/apache/airflow/issues?q=is%3Aissue+utf8mb3+is%3Aclosed

3) Best: start from scratch and start using Postgres. Really, this is the best option. MySQL is by far biggest contriibutor to database problems people have with Airflow. I'd strongly recommend you to choose point 3.

**Reply by cah-peter-g** [reply]
Thank you Jarek! Will forward the same to my team!
